packages/promotion/integration-tests/__tests__/services/promotion-module/compute-actions.spec.ts

```diff
@@ -38,44 +38,40 @@ describe("Promotion Service: computeActions", () => {
   })
 
   describe("when code is not present in database", () => {
-    it("should throw error when code in promotions array does not exist", async () => {
-      const error = await service
-        .computeActions(["DOES_NOT_EXIST"], {
-          customer: {
-            customer_group: {
-              id: "VIP",
+    it("should return empty array when promotion does not exist", async () => {
+      const response = await service.computeActions(["DOES_NOT_EXIST"], {
+        customer: {
+          customer_group: {
+            id: "VIP",
+          },
+        },
+        items: [
+          {
+            id: "item_cotton_tshirt",
+            quantity: 1,
+            unit_price: 100,
+            product_category: {
+              id: "catg_cotton",
+            },
```

Comment:
thought(non-blocking): would probably change this shape to be `product_categories: [{...}]` 

We should do this in a different API but let's add a task to settle on the final shape of the inputs to the promotion module.

Reply:
it works both ways already. You can pass product_categories: [] or product_category: {}. We've only added very few definition restrictions [here](https://github.com/medusajs/medusa/blob/develop/packages/types/src/promotion/common/compute-actions.ts#L64C1-L65C1). 

Let me know if thats not what you meant